packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -143,30 +159,94 @@ contract ControllerHelper is FlashControllerHelper, IERC721Receiver {
             weth,
             wPowerPerp,
             IUniswapV3Pool(wPowerPerpPool).fee(),
             _wPowerPerpAmountToBurn.add(_wPowerPerpAmountToBuy),
-            _maxToPay,
+            _collateralToBuyWith.add(msg.value),
             uint8(FLASH_SOURCE.FLASH_W_BURN),
-            abi.encodePacked(_vaultId, _wPowerPerpAmountToBurn, _wPowerPerpAmountToBuy, _collateralToWithdraw)
+            abi.encodePacked(
+                _vaultId,
+                _wPowerPerpAmountToBurn,
+                _wPowerPerpAmountToBuy,
+                _collateralToWithdraw,
+                _collateralToBuyWith.add(msg.value)
+            )
         );
 
         emit FlashWBurn(msg.sender, _vaultId, _wPowerPerpAmountToBurn, _collateralToWithdraw, _wPowerPerpAmountToBuy);
     }
 
+    /**
+     * @notice mint WPowerPerp and LP into Uniswap v3 pool
+     * @param _vaultId vault ID
+     * @param _wPowerPerpAmount amount of WPowerPerp token to mint
+     * @param _collateralToMint collateral to use for minting
+     * @param _collateralToLP collateral to use for LPing
+     * @param _lowerTick LP lower tick
+     * @param _upperTick LP upper tick
+     */
+    function batchMintLp(
+        uint256 _vaultId,
+        uint256 _wPowerPerpAmount,
+        uint256 _collateralToMint,
+        uint256 _collateralToLP,
+        uint256 _amount0Min,
+        uint256 _amount1Min,
+        uint256 _deadline,
+        int24 _lowerTick,
+        int24 _upperTick
+    ) external payable {
+        require(msg.value == _collateralToMint.add(_collateralToLP), "Wrong ETH sent");
+
+        uint256 vaultId = IController(controller).mintWPowerPerpAmount{value: _collateralToMint}(
+            _vaultId,
+            _wPowerPerpAmount,
+            0
+        );
+        address token0 = IUniswapV3Pool(wPowerPerpPool).token0();
+        address token1 = IUniswapV3Pool(wPowerPerpPool).token1();
+
+        INonfungiblePositionManager.MintParams memory params = INonfungiblePositionManager.MintParams({
+            token0: token0,
+            token1: token1,
+            fee: IUniswapV3Pool(wPowerPerpPool).fee(),
+            tickLower: _lowerTick,
+            tickUpper: _upperTick,
+            amount0Desired: token0 == wPowerPerp ? _wPowerPerpAmount : _collateralToLP,
+            amount1Desired: token1 == wPowerPerp ? _wPowerPerpAmount : _collateralToLP,
+            amount0Min: _amount0Min,
+            amount1Min: _amount1Min,
+            recipient: msg.sender,
+            deadline: _deadline
+        });
+
+        INonfungiblePositionManager(nonfungiblePositionManager).mint{value: _collateralToLP}(params);
+
+        if (_vaultId == 0) IShortPowerPerp(shortPowerPerp).safeTransferFrom(address(this), msg.sender, vaultId);
+        if (address(this).balance > 0) {
+            payable(msg.sender).sendValue(address(this).balance);
+        }
+        uint256 remainingWPowerPerp = IWPowerPerp(wPowerPerp).balanceOf(address(this));
+        if (remainingWPowerPerp > 0) {
```

Comment:
should we just burn it vs returning it?

Reply:
seems more annoying to get some back and have to manually burn it or sell it or something

Reply:
sounds good, will burn it

Reply:
But not sure if it is worth to do it here as I need to get the amount of collateral to withdraw, and require calling oracle..etc which maybe not that efficient, what u think ?

Reply:
Thought on just burning the debt without withdrawing collateral ?